lib/Backends/CPU/CPUDeviceManager.cpp

```diff
@@ -108,6 +120,12 @@ void CPUDeviceManager::addNetworkImpl(const Module *module,
       func.second->getRuntimeBundle().collectConstants(module);
     }
     functions_.emplace(func.first, func.second);
+
+    auto &bundle = func.second->getRuntimeBundle();
+    std::unique_ptr<CPUBuffer> deviceBuffer = llvm::make_unique<CPUBuffer>(
```

Comment:
Does it mean that we allocate buffers for activations and mutable weights at the network deployment time? Shouldn't this happen per run of the deployed network? Or may be I miss something?

Reply:
Yes.
This is a point worth dwelling on because this is the point of this PR - it seems that the CPU does this allocation on execution while other backends (openCL, Habana) allocates on deployment. It's a time-memory tradeoff.
It is separated from the follow-up issue I was about to address - which is the actual loading and updating of the placeholders - which really doesn't have to happen for all placeholders each run. 

For example on Habana:
https://github.com/pytorch/glow/blob/63e16c654ee872c9b4539165cad71e20e5a66656/lib/Backends/Habana/HabanaDeviceManager.cpp#L186
https://github.com/pytorch/glow/blob/63e16c654ee872c9b4539165cad71e20e5a66656/lib/Backends/Habana/HabanaFunction.cpp#L61

Reply:
How would you run multiple instances of the same network concurrently if there is only one buffer for activations and mutable weights? IMHO, you can share the buffer for constants among multiple running instances of the same network, but you cannot share the memory allocated for activations and mutable weights. Glow's CPU backend was able to run multiple instances of the same network concurrently without any issues using the existing buffer allocation scheme IIRC. But it wouldn't be possible anymore after your change, or?

Reply:
True. If the same DeviceManager instance is used, I wouldn't be able to run a few requests concurrently. Let me check this.

Reply:
@gcatron Are DeviceManagers supposed to support execution of a few instances of the same NN model concurrently? If so, does it mean that OpenCL and Habana do not support it yet based on what @mortzur said above?

Reply:
For any of the DeviceManagers that inherit from QueueBacked, (OpenCL, Interpreter, and CPU) they use a threadpool of size 1 to run their requests which mean they don't support concurrent execution. Requests can be queued concurrently but only one is executed at a time.

Reply:
The solution for openCL and Habana device managers is to use buffer pools (circular buffer in both cases I think) to enable k-parallel requests. I can do a similar thing here.